src/Forms/GridField/GridField.php

```diff
@@ -548,6 +549,12 @@ public function FieldHolder($properties = [])
                     }
                 }
             }
+            if ($item instanceof GridFieldPaginator) {
+                $parameters = $item->getTemplateParameters($this);
+                if ($parameters) {
+                    $total = $parameters->NumRecords;
+                }
+            }
```

Comment:
I'm not convinced it's worth running all of the logic in `getTemplateParameters()` just for this - I'd be more comfortable with a `getTotalItems()` method being added to `GridFieldPaginator` in a minor release, which can then be called here.

I will note the beta release for CMS 5.4 (which is also the feature freeze for all of CMS 5) is currently scheduled for 17 Feb, so if you want to make that change, it has to be merged by then.

Reply:
yes, i fully agree. i was a bit unsure about adding a new method but i'm happy to add it and adjust to PR